packages/console/src/components/MultilineInput/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { nanoid } from 'nanoid';
+import React, { FormEvent, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import * as textButtonStyles from '@/components/TextButton/index.module.scss';
+import Minus from '@/icons/Minus';
+
+import IconButton from '../IconButton';
+import TextInput from '../TextInput';
+import * as styles from './index.module.scss';
+
+type Props = {
+  values: string[];
+  onChange: (values: string[]) => void;
+};
+
+type FieldItem = {
+  id: string;
+  value: string;
+};
+
+const MultilineInput = ({ values, onChange }: Props) => {
+  const [fields, setFileds] = useState<FieldItem[]>([]);
+
+  const { t } = useTranslation(undefined, {
+    keyPrefix: 'general',
+  });
+
+  useEffect(() => {
```

Comment:
Can be optimized by using `useMemo`.

```
const fields = useMemo(xxx, [values]);
```

and call `onChange` directly instead of `setFields`

Reply:
updated